### PR TITLE
Adding build material for windows dlls and updating samples

### PR DESCRIPTION
**Pull request by jplafonta**
* Add project files for libHttpClient.GDK.dll and libHttpClient.Win32.dll
* Add props file libHttpClient.import.props which adds references to the appropriate lhc projects.  By default, it will add a reference to the dll project, but this can be overridden by specifying the build property HCStaticLib=true prior to importing the props file.  The existing libHttpClient.props file will continue to add references to the old static lib projects as to not break existing clients.

### CODE REVIEW

File: Build/libHttpClient.GDK/libHttpClient.GDK.log [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿  Building against GXDK Edition 231001.
```
Comment (jasonsandlin) [inline Build/libHttpClient.GDK/libHttpClient.GDK.log:1]: .gitignore log files

File: Build/libHttpClient.Win32/libHttpClient.Win32.log [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿  uri.cpp
```
Comment (jasonsandlin) [inline Build/libHttpClient.Win32/libHttpClient.Win32.log:1]: remove

File: Include/XAsync.h [hunk 3]
```diff
@@ -69,7 +69,7 @@ struct XAsyncBlock
 /// </summary>
 /// <param name='asyncBlock'>A pointer to the XAsyncBlock that was passed to the async call.</param>
 /// <param name='wait'>If true, XAsyncGetStatus waits until the async call either completes or is canceled.</param>
-STDAPI XAsyncGetStatus(
+HC_API XAsyncGetStatus(
```
Comment (jasonsandlin) [inline Include/XAsync.h:72]: XAsync* is shared source from Brian, so changing it might not be wise

**Review by jasonsandlin — APPROVED** [review]

File: Include/XAsync.h (hunk 3 above)
Comment (jplafonta) [reply Include/XAsync.h:72]: Good point.  Another option would be to export the XAsync APIs (or even all the LHC APIs) with a .def file.  I find that to be slightly more annoying to maintain but maybe its worth it to leave the source untouched in this case.  Thoughts?

Comment (jasonsandlin) [reply Include/XAsync.h:72]: def file is better.  this api list doesn't change often. i'd rather be able to keep the source in sync.

**Pull request merged**